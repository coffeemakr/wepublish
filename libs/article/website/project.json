{
  "name": "article-website",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/article/website/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/article/website/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/article/website",
        "tsConfig": "libs/article/website/tsconfig.lib.json",
        "packageJson": "libs/article/website/package.json",
        "main": "libs/article/website/src/index.ts",
        "assets": ["libs/article/website/*.md"],
        "updateBuildableProjectDepsInPackageJson": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/article/website/jest.config.ts",
        "passWithNoTests": true
      }
    }
  }
}
