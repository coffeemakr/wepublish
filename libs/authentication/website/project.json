{
  "name": "authentication-website",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/authentication/website/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/authentication/website/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/authentication/website",
        "tsConfig": "libs/authentication/website/tsconfig.lib.json",
        "packageJson": "libs/authentication/website/package.json",
        "main": "libs/authentication/website/src/index.ts",
        "assets": ["libs/authentication/website/*.md"],
        "updateBuildableProjectDepsInPackageJson": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/authentication/website/jest.config.ts",
        "passWithNoTests": true
      }
    }
  }
}
