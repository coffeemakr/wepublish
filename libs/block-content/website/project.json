{
  "name": "block-content-website",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/block-content/website/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/block-content/website/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/block-content/website",
        "tsConfig": "libs/block-content/website/tsconfig.lib.json",
        "packageJson": "libs/block-content/website/package.json",
        "main": "libs/block-content/website/src/index.ts",
        "assets": ["libs/block-content/website/*.md"],
        "updateBuildableProjectDepsInPackageJson": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/block-content/website/jest.config.ts",
        "passWithNoTests": true
      }
    }
  }
}
