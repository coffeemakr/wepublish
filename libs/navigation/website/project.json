{
  "name": "navigation-website",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/navigation/website/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/navigation/website/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/navigation/website",
        "tsConfig": "libs/navigation/website/tsconfig.lib.json",
        "packageJson": "libs/navigation/website/package.json",
        "main": "libs/navigation/website/src/index.ts",
        "assets": ["libs/navigation/website/*.md"],
        "updateBuildableProjectDepsInPackageJson": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/navigation/website/jest.config.ts",
        "passWithNoTests": true
      }
    }
  }
}
