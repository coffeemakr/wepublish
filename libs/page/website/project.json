{
  "name": "page-website",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/page/website/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/page/website/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/page/website",
        "tsConfig": "libs/page/website/tsconfig.lib.json",
        "packageJson": "libs/page/website/package.json",
        "main": "libs/page/website/src/index.ts",
        "assets": ["libs/page/website/*.md"],
        "updateBuildableProjectDepsInPackageJson": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/page/website/jest.config.ts",
        "passWithNoTests": true
      }
    }
  }
}
