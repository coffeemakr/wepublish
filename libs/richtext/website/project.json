{
  "name": "richtext-website",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/richtext/website/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/richtext/website/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/richtext/website",
        "tsConfig": "libs/richtext/website/tsconfig.lib.json",
        "packageJson": "libs/richtext/website/package.json",
        "main": "libs/richtext/website/src/index.ts",
        "assets": ["libs/richtext/website/*.md"],
        "updateBuildableProjectDepsInPackageJson": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/richtext/website/jest.config.ts",
        "passWithNoTests": true
      }
    }
  }
}
