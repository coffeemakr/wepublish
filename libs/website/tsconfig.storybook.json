{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "emitDecoratorMetadata": true,
    "outDir": ""
  },
  "files": [
    "../../node_modules/@nx/react/typings/styled-jsx.d.ts",
    "../../node_modules/@nx/react/typings/cssmodule.d.ts",
    "../../node_modules/@nx/react/typings/image.d.ts"
  ],
  "exclude": ["src/**/*.spec.ts", "src/**/*.spec.js", "src/**/*.spec.tsx", "src/**/*.spec.jsx"],
  "include": [
    "../**/website/src/**/*.stories.ts",
    "../**/website/src/**/*.stories.js",
    "../**/website/src/**/*.stories.jsx",
    "../**/website/src/**/*.stories.tsx",
    "../**/website/src/**/*.stories.mdx",
    ".storybook/*.js",
    "preview.tsx",
    "main.ts"
  ]
}
