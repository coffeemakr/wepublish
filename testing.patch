diff --git a/libs/api/src/lib/db/article.ts b/libs/api/src/lib/db/article.ts
index 7359af6a..ea27be09 100644
--- a/libs/api/src/lib/db/article.ts
+++ b/libs/api/src/lib/db/article.ts
@@ -14,11 +14,11 @@ export interface ArticleData {
   readonly lead?: string | null
   readonly seoTitle?: string | null
   readonly slug?: string | null
-  readonly tags: string[]
+  readonly tags?: string[]
 
   readonly canonicalUrl?: string | null
 
-  readonly properties: MetadataProperty[]
+  readonly properties?: MetadataProperty[]
 
   readonly imageID?: string | null
   readonly authors: ArticleRevisionAuthor[]
@@ -107,7 +107,7 @@ export type ArticleRevisionWithRelations = PrismaArticleRevision & {
 
 export type ArticleWithRevisions = PrismaArticle & {
   draft: ArticleRevisionWithRelations | null
-  pending: ArticleRevisionWithRelations | null
+  pending?: ArticleRevisionWithRelations | null
   published: ArticleRevisionWithRelations | null
 }
 
diff --git a/libs/api/src/lib/db/page.ts b/libs/api/src/lib/db/page.ts
index fb73c289..6f0ed1b5 100644
--- a/libs/api/src/lib/db/page.ts
+++ b/libs/api/src/lib/db/page.ts
@@ -14,9 +14,9 @@ export interface PageData {
 
   readonly title: string
   readonly description?: string | null
-  readonly tags: string[]
+  readonly tags?: string[]
 
-  readonly properties: MetadataProperty[]
+  readonly properties?: MetadataProperty[]
 
   readonly imageID?: string | null
 
@@ -85,7 +85,7 @@ export type PageRevisionWithProperties = PrismaPageRevision & {
 
 export type PageWithRevisions = PrismaPage & {
   draft: PageRevisionWithProperties | null
-  pending: PageRevisionWithProperties | null
+  pending?: PageRevisionWithProperties | null
   published: PageRevisionWithProperties | null
 }
 
diff --git a/libs/api/src/lib/graphql/article.ts b/libs/api/src/lib/graphql/article.ts
index 8d80fb27..77357e01 100644
--- a/libs/api/src/lib/graphql/article.ts
+++ b/libs/api/src/lib/graphql/article.ts
@@ -275,7 +275,7 @@ export const GraphQLPublicArticle: GraphQLObjectType<PublicArticle, Context> =
         type: GraphQLNonNull(GraphQLList(GraphQLNonNull(GraphQLMetadataPropertyPublic))),
         resolve: ({properties}) => {
           return properties
-            .filter(property => property.public)
+            ?.filter(property => property.public)
             .map(({key, value}) => ({key, value}))
         }
       },
diff --git a/libs/api/src/lib/graphql/article/article.private-mutation.ts b/libs/api/src/lib/graphql/article/article.private-mutation.ts
index 43a6a3ee..049c7e54 100644
--- a/libs/api/src/lib/graphql/article/article.private-mutation.ts
+++ b/libs/api/src/lib/graphql/article/article.private-mutation.ts
@@ -178,7 +178,8 @@ export const duplicateArticle = async (
       createMany: {
         data: socialMediaAuthors.map(({authorId}) => ({authorId}))
       }
-    }
+    },
+    blocks: []
   }
 
   return articleClient.create({
@@ -303,7 +304,8 @@ export const unpublishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -333,7 +335,8 @@ export const unpublishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           }
         }
       },
@@ -491,7 +494,8 @@ export const publishArticle = async (
                 createMany: {
                   data: socialMediaAuthors.map(({authorId}) => ({authorId}))
                 }
-              }
+              },
+              blocks: []
             },
             update: {
               ...revision,
@@ -521,7 +525,8 @@ export const publishArticle = async (
                 createMany: {
                   data: socialMediaAuthors.map(({authorId}) => ({authorId}))
                 }
-              }
+              },
+              blocks: []
             }
           }
         },
@@ -579,7 +584,8 @@ export const publishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -609,7 +615,8 @@ export const publishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           }
         }
       },
diff --git a/libs/api/src/lib/graphql/page.ts b/libs/api/src/lib/graphql/page.ts
index 56d380f0..6b63518e 100644
--- a/libs/api/src/lib/graphql/page.ts
+++ b/libs/api/src/lib/graphql/page.ts
@@ -210,7 +210,7 @@ export const GraphQLPublicPage = new GraphQLObjectType<PublicPage, Context>({
     properties: {
       type: GraphQLNonNull(GraphQLList(GraphQLNonNull(GraphQLMetadataPropertyPublic))),
       resolve: ({properties}) => {
-        return properties.filter(property => property.public).map(({key, value}) => ({key, value}))
+        return properties?.filter(property => property.public).map(({key, value}) => ({key, value}))
       }
     },
 
diff --git a/libs/api/src/lib/graphql/page/page.private-mutation.ts b/libs/api/src/lib/graphql/page/page.private-mutation.ts
index 8d21a995..456df84e 100644
--- a/libs/api/src/lib/graphql/page/page.private-mutation.ts
+++ b/libs/api/src/lib/graphql/page/page.private-mutation.ts
@@ -140,7 +140,8 @@ export const duplicatePage = async (
       createMany: {
         data: duplicatedProperties
       }
-    }
+    },
+    blocks: []
   }
 
   return pageClient.create({
@@ -236,7 +237,8 @@ export const unpublishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -250,7 +252,8 @@ export const unpublishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           }
         }
       },
@@ -387,7 +390,8 @@ export const publishPage = async (
                 createMany: {
                   data: properties
                 }
-              }
+              },
+              blocks: []
             },
             update: {
               ...revision,
@@ -401,7 +405,8 @@ export const publishPage = async (
                 createMany: {
                   data: properties
                 }
-              }
+              },
+              blocks: []
             }
           }
         },
@@ -443,7 +448,8 @@ export const publishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -454,7 +460,8 @@ export const publishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           }
         }
       },
