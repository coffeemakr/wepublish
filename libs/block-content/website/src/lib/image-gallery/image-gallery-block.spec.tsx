import {runStorybookTests} from '@wepublish/testing/website'
import * as stories from './image-gallery-block.stories'

describe('Image Gallery Block', () => {
  runStorybookTests(stories)
})
