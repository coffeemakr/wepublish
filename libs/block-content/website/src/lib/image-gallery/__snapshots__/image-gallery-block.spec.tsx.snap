// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Gallery Block should render Default 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
}

.emotion-3 {
  display: grid;
  overflow-y: auto;
  list-style: none;
  padding: 0;
  -webkit-overflow-scrolling: touch;
  margin: 0;
  grid-template-columns: repeat(1, 1fr);
}

@media (min-width:600px) {
  .emotion-3 {
    grid-template-columns: repeat(2, 1fr);
  }
}

@media (min-width:900px) {
  .emotion-3 {
    grid-template-columns: repeat(3, 1fr);
  }
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-4 .MuiImageListItem-img {
  object-fit: cover;
  width: 100%;
  height: auto;
  display: block;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-5 {
  max-width: 100%;
  height: auto;
  aspect-ratio: auto 1;
  max-height: 80lvh;
  object-fit: contain;
}

.emotion-7 {
  position: relative;
  left: 0;
  right: 0;
  background: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
}

.emotion-8 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 6px 0 12px;
  color: inherit;
  overflow: hidden;
}

.emotion-9 {
  font-size: 1rem;
  line-height: 24px;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

<figure
    class="emotion-0 emotion-1"
  >
    <ul
      class="MuiImageList-root MuiImageList-standard emotion-2 emotion-3"
      style="grid-template-columns: repeat(0, 1fr); gap: 8px;"
    >
      <li
        class="MuiImageListItem-root MuiImageListItem-standard emotion-4"
        style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
      >
        <img
          alt="An image description"
          class="emotion-5 emotion-6"
          loading="lazy"
          srcset="https://unsplash.it/800/800?0 800w,
https://unsplash.it/500/500?0 500w,
https://unsplash.it/300/300?0 300w,
https://unsplash.it/200/200?0 200w"
          title="An image title"
        />
        <div
          class="MuiImageListItemBar-root MuiImageListItemBar-positionBelow emotion-7"
        >
          <div
            class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBelow emotion-8"
          >
            <div
              class="MuiImageListItemBar-title emotion-9"
            >
              Image caption
            </div>
          </div>
        </div>
      </li>
      <li
        class="MuiImageListItem-root MuiImageListItem-standard emotion-4"
        style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
      >
        <img
          alt="An image description"
          class="emotion-5 emotion-6"
          loading="lazy"
          srcset="https://unsplash.it/800/800?1 800w,
https://unsplash.it/500/500?1 500w,
https://unsplash.it/300/300?1 300w,
https://unsplash.it/200/200?1 200w"
          title="An image title"
        />
        <div
          class="MuiImageListItemBar-root MuiImageListItemBar-positionBelow emotion-7"
        >
          <div
            class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBelow emotion-8"
          >
            <div
              class="MuiImageListItemBar-title emotion-9"
            >
              Image caption
            </div>
          </div>
        </div>
      </li>
      <li
        class="MuiImageListItem-root MuiImageListItem-standard emotion-4"
        style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
      >
        <img
          alt="An image description"
          class="emotion-5 emotion-6"
          loading="lazy"
          srcset="https://unsplash.it/800/800?2 800w,
https://unsplash.it/500/500?2 500w,
https://unsplash.it/300/300?2 300w,
https://unsplash.it/200/200?2 200w"
          title="An image title"
        />
        <div
          class="MuiImageListItemBar-root MuiImageListItemBar-positionBelow emotion-7"
        >
          <div
            class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBelow emotion-8"
          >
            <div
              class="MuiImageListItemBar-title emotion-9"
            >
              Image caption
            </div>
          </div>
        </div>
      </li>
      <li
        class="MuiImageListItem-root MuiImageListItem-standard emotion-4"
        style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
      >
        <img
          alt="An image description"
          class="emotion-5 emotion-6"
          loading="lazy"
          srcset="https://unsplash.it/800/800?3 800w,
https://unsplash.it/500/500?3 500w,
https://unsplash.it/300/300?3 300w,
https://unsplash.it/200/200?3 200w"
          title="An image title"
        />
        <div
          class="MuiImageListItemBar-root MuiImageListItemBar-positionBelow emotion-7"
        >
          <div
            class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBelow emotion-8"
          >
            <div
              class="MuiImageListItemBar-title emotion-9"
            >
              Image caption
            </div>
          </div>
        </div>
      </li>
      <li
        class="MuiImageListItem-root MuiImageListItem-standard emotion-4"
        style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
      >
        <img
          alt="An image description"
          class="emotion-5 emotion-6"
          loading="lazy"
          srcset="https://unsplash.it/800/800?4 800w,
https://unsplash.it/500/500?4 500w,
https://unsplash.it/300/300?4 300w,
https://unsplash.it/200/200?4 200w"
          title="An image title"
        />
        <div
          class="MuiImageListItemBar-root MuiImageListItemBar-positionBelow emotion-7"
        >
          <div
            class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBelow emotion-8"
          >
            <div
              class="MuiImageListItemBar-title emotion-9"
            >
              Image caption
            </div>
          </div>
        </div>
      </li>
      <li
        class="MuiImageListItem-root MuiImageListItem-standard emotion-4"
        style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
      >
        <img
          alt="An image description"
          class="emotion-5 emotion-6"
          loading="lazy"
          srcset="https://unsplash.it/800/800?5 800w,
https://unsplash.it/500/500?5 500w,
https://unsplash.it/300/300?5 300w,
https://unsplash.it/200/200?5 200w"
          title="An image title"
        />
        <div
          class="MuiImageListItemBar-root MuiImageListItemBar-positionBelow emotion-7"
        >
          <div
            class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBelow emotion-8"
          >
            <div
              class="MuiImageListItemBar-title emotion-9"
            >
              Image caption
            </div>
          </div>
        </div>
      </li>
      <li
        class="MuiImageListItem-root MuiImageListItem-standard emotion-4"
        style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
      >
        <img
          alt="An image description"
          class="emotion-5 emotion-6"
          loading="lazy"
          srcset="https://unsplash.it/800/800?6 800w,
https://unsplash.it/500/500?6 500w,
https://unsplash.it/300/300?6 300w,
https://unsplash.it/200/200?6 200w"
          title="An image title"
        />
        <div
          class="MuiImageListItemBar-root MuiImageListItemBar-positionBelow emotion-7"
        >
          <div
            class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBelow emotion-8"
          >
            <div
              class="MuiImageListItemBar-title emotion-9"
            >
              Image caption
            </div>
          </div>
        </div>
      </li>
      <li
        class="MuiImageListItem-root MuiImageListItem-standard emotion-4"
        style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
      >
        <img
          alt="An image description"
          class="emotion-5 emotion-6"
          loading="lazy"
          srcset="https://unsplash.it/800/800?7 800w,
https://unsplash.it/500/500?7 500w,
https://unsplash.it/300/300?7 300w,
https://unsplash.it/200/200?7 200w"
          title="An image title"
        />
        <div
          class="MuiImageListItemBar-root MuiImageListItemBar-positionBelow emotion-7"
        >
          <div
            class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBelow emotion-8"
          >
            <div
              class="MuiImageListItemBar-title emotion-9"
            >
              Image caption
            </div>
          </div>
        </div>
      </li>
      <li
        class="MuiImageListItem-root MuiImageListItem-standard emotion-4"
        style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
      >
        <img
          alt="An image description"
          class="emotion-5 emotion-6"
          loading="lazy"
          srcset="https://unsplash.it/800/800?8 800w,
https://unsplash.it/500/500?8 500w,
https://unsplash.it/300/300?8 300w,
https://unsplash.it/200/200?8 200w"
          title="An image title"
        />
        <div
          class="MuiImageListItemBar-root MuiImageListItemBar-positionBelow emotion-7"
        >
          <div
            class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBelow emotion-8"
          >
            <div
              class="MuiImageListItemBar-title emotion-9"
            >
              Image caption
            </div>
          </div>
        </div>
      </li>
      <li
        class="MuiImageListItem-root MuiImageListItem-standard emotion-4"
        style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
      >
        <img
          alt="An image description"
          class="emotion-5 emotion-6"
          loading="lazy"
          srcset="https://unsplash.it/800/800?9 800w,
https://unsplash.it/500/500?9 500w,
https://unsplash.it/300/300?9 300w,
https://unsplash.it/200/200?9 200w"
          title="An image title"
        />
        <div
          class="MuiImageListItemBar-root MuiImageListItemBar-positionBelow emotion-7"
        >
          <div
            class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBelow emotion-8"
          >
            <div
              class="MuiImageListItemBar-title emotion-9"
            >
              Image caption
            </div>
          </div>
        </div>
      </li>
      <li
        class="MuiImageListItem-root MuiImageListItem-standard emotion-4"
        style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
      >
        <img
          alt="An image description"
          class="emotion-5 emotion-6"
          loading="lazy"
          srcset="https://unsplash.it/800/800?10 800w,
https://unsplash.it/500/500?10 500w,
https://unsplash.it/300/300?10 300w,
https://unsplash.it/200/200?10 200w"
          title="An image title"
        />
        <div
          class="MuiImageListItemBar-root MuiImageListItemBar-positionBelow emotion-7"
        >
          <div
            class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBelow emotion-8"
          >
            <div
              class="MuiImageListItemBar-title emotion-9"
            >
              Image caption
            </div>
          </div>
        </div>
      </li>
    </ul>
  </figure>
</DocumentFragment>
`;

exports[`Image Gallery Block should render WithClassName 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,8 +1,8 @@
  <DocumentFragment>
    <figure
-     class="css-1b0x59d-ImageGalleryBlockWrapper e5naayd1"
+     class="extra-classname css-1b0x59d-ImageGalleryBlockWrapper e5naayd1"
    >
      <ul
        class="MuiImageList-root MuiImageList-standard e5naayd0 css-eeca2y-MuiImageList-root-ImageGalleryBlockImageList"
        style="grid-template-columns: repeat(0, 1fr); gap: 8px;"
      >"
`;

exports[`Image Gallery Block should render WithEmotion 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Image Gallery Block should render WithoutCaption 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -26,13 +26,11 @@
            <div
              class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBelow css-186gwnw-MuiImageListItemBar-titleWrap"
            >
              <div
                class="MuiImageListItemBar-title css-dasnyc-MuiImageListItemBar-title"
-             >
-               Image caption
-             </div>
+             />
            </div>
          </div>
        </li>
        <li
          class="MuiImageListItem-root MuiImageListItem-standard css-ta95xa-MuiImageListItem-root"
@@ -194,13 +192,11 @@
            <div
              class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBelow css-186gwnw-MuiImageListItemBar-titleWrap"
            >
              <div
                class="MuiImageListItemBar-title css-dasnyc-MuiImageListItemBar-title"
-             >
-               Image caption
-             </div>
+             />
            </div>
          </div>
        </li>
        <li
          class="MuiImageListItem-root MuiImageListItem-standard css-ta95xa-MuiImageListItem-root""
`;
